File: myrana_flutter/android/app/src/main/kotlin/com/example/myrana/core/ChildCodeNormalizer.kt
package com.example.myrana.core

/**
 * مطابق لـ ChildCodeNormalizer في Flutter:
 * - تنظيف: إزالة CHILD- / الفراغات / _ / -
 * - تحويل إلى uppercase
 * - إزالة أي محارف غير [A-Z0-9]
 * - إرجاع without prefix (كما يتوقع السيرفر عبر db_child_code)
 */
object ChildCodeNormalizer {

    fun clean(raw: String): String {
        var s = raw.trim().uppercase().replace(Regex("[\\s\\-_]"), "")
        if (s.startsWith("CHILD")) s = s.substring(5)
        return s.replace(Regex("[^A-Z0-9]"), "")
    }

    /** كما يرسلها السيرفر/Flutter للتخزين (بدون بادئة CHILD). */
    fun forApi(raw: String): String = clean(raw)
}

File: myrana_flutter/lib/util/child_code_normalizer.dart
/// مطابق لـ ChildCodeNormalizer.kt
class ChildCodeNormalizer {
  ChildCodeNormalizer._();

  static String clean(String raw) {
    var s = raw.trim().toUpperCase().replaceAll(RegExp(r'[\s\-_]'), '');
    if (s.startsWith('CHILD')) s = s.substring(5);
    return s.replaceAll(RegExp(r'[^A-Z0-9]'), '');
  }

  static String normalize(String raw) {
    final c = clean(raw);
    if (c.isEmpty) return '';
    return c.startsWith('CHILD') ? c : 'CHILD$c';
  }

  /// كما يرسله تطبيق Kotlin للسيرفر (بدون بادئة CHILD في التخزين).
  static String forApi(String raw) => clean(raw);
}
